packages/query/src/index.ts

```diff
@@ -721,10 +724,66 @@ const getQueryFnArguments = ({
   return '{ signal }';
 };
 
+const generateQueryKeyImplementation = ({
+  query,
+  body,
+  outputClient,
+  route,
+  queryParams,
+  props,
+  shouldSplitQueryKey,
+  shouldExportQueryKey,
+}: {
+  query: {
+    name: string;
+    options: unknown;
+    type: QueryType;
+    queryParam?: string;
+    queryKeyFnName: string;
+  };
+  body: GetterBody;
+  outputClient: OutputClient | OutputClientFunc;
+  route: GeneratorVerbOptions['route'];
+  queryParams: GeneratorVerbOptions['queryParams'];
+  props: GeneratorVerbOptions['props'];
+  shouldSplitQueryKey: NormalizedOverrideOutput['query']['shouldSplitQueryKey'];
+  shouldExportQueryKey: NormalizedOverrideOutput['query']['shouldExportQueryKey'];
+}) => {
+  const queryKeyProps = makeParamsOptional(
+    toObjectString(
+      props.filter((prop) => prop.type !== GetterPropType.HEADER),
+      'implementation',
+    ),
+  );
+
+  const routeString =
+    isVue(outputClient) || shouldSplitQueryKey
+      ? getRouteAsArray(route) // Note: this is required for reactivity to work, we will lose it if route params are converted into string, only as array they will be tracked // TODO: add tests for this
+      : `\`${route}\``;
+
+  // Note: do not unref() params in Vue - this will make key lose reactivity
+  return `
+
+${shouldExportQueryKey ? 'export ' : ''}const ${query.queryKeyFnName} = (${queryKeyProps}) => {
+    return [
+    ${[
+      routeString,
+      query.type === QueryType.INFINITE ||
+      query.type === QueryType.SUSPENSE_INFINITE
+        ? `'infinite'`
+        : '',
+      queryParams ? '...(params ? [params]: [])' : '',
+      body.implementation,
+    ]
+      .filter((x) => !!x)
+      .join(', ')}
+    ] as const;
```

Comment:
Wouldn't it be enough to just prepend the string to the array for `infinate`?

Reply:
Great, then let's just add a prefix for a more concise fix.

Reply:
Added the prefix as you suggested